src/app-layout/__tests__/trigger-button.test.tsx

```diff
@@ -0,0 +1,353 @@
+  return { wrapper, rerender, getByTestId, getByText };
+};
+
+const renderVisualRefreshToolbarTriggerButton = (
+  props: Partial<VisualRefreshToolbarTriggerButtonProps> = {},
+  ref: React.Ref<ButtonProps.Ref>
+) => {
+  const renderProps = { ...mockProps, ...props };
+  const { container, rerender, getByTestId, getByText } = render(
+    <div>
+      <VisualRefreshToolbarTriggerButton {...(renderProps as VisualRefreshToolbarTriggerButtonProps)} ref={ref} />
+      <span className={mockOtherEl.class}>{mockOtherEl.text}</span>
+    </div>
+  );
+  const wrapper = createWrapper(container).findByClassName(toolbarTriggerButtonStyles['trigger-wrapper'])!;
+  return { wrapper, rerender, getByTestId, getByText };
+};
+
+describe('Visual refresh trigger-button (not in appLayoutWidget toolbar)', () => {
+  beforeEach(() => jest.clearAllMocks());
+
+  //tests for both desktop and mobile
+  describe.each([true, false])('Toolbar trigger-button with isMobile=%s', isMobile => {
+    //tests for both drawer being opend and closed
+    describe.each([true, false])('AppLayoutInternals with hasOpenDrawer=%s', hasOpenDrawer => {
+      test('renders correctly with without badge', () => {
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const { wrapper, getByTestId } = renderVisualRefreshTriggerButton(
+          {
+            badge: false,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref
+        );
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+        expect(wrapper!.findIcon()).toBeTruthy();
+        expect(wrapper.findByClassName(visualRefreshStyles.dot)).toBeNull();
+        expect(wrapper.findBadge()).toBeNull();
+      });
+
+      test('renders correctly with aria controls', () => {
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const mockAriaControls = 'mock-aria-control';
+        const { wrapper, getByTestId } = renderVisualRefreshTriggerButton(
+          {
+            badge: false,
+            ariaControls: mockAriaControls,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref
+        );
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+        expect(button!.getElement().getAttribute('aria-controls')).toEqual(mockAriaControls);
+      });
+
+      test('renders correctly with iconName', () => {
```

Comment:
I didn't get this test. Is icon name assigned implicitly when the badge is not provided?

Reply:
The asserts also look very similar to the previous tests, I wonder if there is a value of keeping this one.